### PR TITLE
Fixes a Bunch of Initializes

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Did some basic testing, this seems to fix initialzing not handling `qdel` and some other things.

I'd like this tested on the liveserver to make sure it doesn't break anything

I knew us continuing to use `New` would bite us in the butt one day.

:cl: Fox McCloud
fix: Fixes some initializations
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
We knew that for a while but we also knew that would invovle removing all the news...AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

....AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by Fox-McCloud** [comment]
I'd like to note, this doesn't fix every single problem.

This fixes 95% of the mobs (exceptions being observers and pAIs) and ensures anything that uses `INITIALIZE_HINT_QDEL` *SHOULD* work.

Every single non-datum `New` needs to be purged, one day, though.

**Comment by ChewedPencil** [comment]
Think i found the first bug. was a robotic brain and was placed in a borg but it wont actually initialize me. cant choose a name and am blind and inoperative

**Comment by Kyep** [comment]
Runtimes (the robot ones and table rack ones being common):

[2019-09-01T23:29:21] Runtime in robot.dm,805: Cannot execute null.start().
   proc name: attacked by (/mob/living/silicon/robot/attacked_by)
   usr: Steven Farlll (sheppysneik) (/mob/living/carbon/human)
   usr.loc: The floor (175,121,1) (/turf/simulated/floor/plasteel)
   src: Default Robot-0 (/mob/living/silicon/robot)
   src.loc: the floor (174,121,1) (/turf/simulated/floor/plasteel)
   call stack:
   Default Robot-0 (/mob/living/silicon/robot): attacked by(the wrench (/obj/item/wrench), Steven Farlll (/mob/living/carbon/human), null)

[2019-09-01T23:27:36] Runtime in robot.dm,733: Cannot execute null.Interact().
   proc name: attackby (/mob/living/silicon/robot/attackby)
   usr: Steven Farlll (sheppysneik) (/mob/living/carbon/human)
   usr.loc: The floor (175,120,1) (/turf/simulated/floor/plasteel)
   src: Default Robot-0 (/mob/living/silicon/robot)
   src.loc: the floor (174,121,1) (/turf/simulated/floor/plasteel)
   call stack:
   Default Robot-0 (/mob/living/silicon/robot): attackby(the wirecutters (/obj/item/wirecutters), Steven Farlll (/mob/living/carbon/human), "icon-x=21;icon-y=19;left=1;scr...")

[2019-09-01T23:32:56] Runtime in robot_defense.dm,50: Cannot modify null.installed.
   proc name: attack hand (/mob/living/silicon/robot/attack_hand)
   usr: Agamemnon (mrmagolor) (/mob/living/carbon/human)
   usr.loc: The floor (175,122,1) (/turf/simulated/floor/plasteel)
   src: Default Robot-0 (/mob/living/silicon/robot)
   src.loc: the floor (175,121,1) (/turf/simulated/floor/plasteel)
   call stack:
   Default Robot-0 (/mob/living/silicon/robot): attack hand(Agamemnon (/mob/living/carbon/human))

[2019-09-01T23:23:46] Runtime in robot_parts.dm,303: Cannot modify null.wrapped.
   proc name: attackby (/obj/item/robot_parts/robot_suit/attackby)
   usr: Agamemnon (mrmagolor) (/mob/living/carbon/human)
   usr.loc: The floor (174,122,1) (/turf/simulated/floor/plasteel)
   src: the robot endoskeleton (/obj/item/robot_parts/robot_suit)
   src.loc: the floor (174,121,1) (/turf/simulated/floor/plasteel)
   call stack:
   the robot endoskeleton (/obj/item/robot_parts/robot_suit): attackby(The robotic brain (SINA-574) (/obj/item/mmi/robotic_brain), Agamemnon (/mob/living/carbon/human), "icon-x=14;icon-y=21;left=1;scr...")

[2019-09-01T23:19:48] Runtime in tables_racks.dm,94: Cannot read null.pass_flags
   proc name: attack hand (/obj/structure/table/attack_hand)
   usr: Landen Philips (lalchi) (/mob/living/carbon/human)
   usr.loc: The floor (113,80,1) (/turf/simulated/floor/plasteel)
   src: the reinforced table (/obj/structure/table/reinforced)
   src.loc: the floor (113,79,1) (/turf/simulated/floor/plasteel)
   call stack:
   the reinforced table (/obj/structure/table/reinforced): attack hand(Landen Philips (/mob/living/carbon/human))
   Landen Philips (/mob/living/carbon/human): UnarmedAttack(the reinforced table (/obj/structure/table/reinforced), 1)
   Landen Philips (/mob/living/carbon/human): ClickOn(the reinforced table (/obj/structure/table/reinforced), "icon-x=8;icon-y=9;left=1;scree...")

[2019-09-01T23:19:36] Runtime in outfit.dm,111: bad number
   proc name: equip (/datum/outfit/proc/equip)
   src: Clown (/datum/outfit/job/clown)
   call stack:
   Clown (/datum/outfit/job/clown): equip(Elliot Black (/mob/living/carbon/human), 0)
   Elliot Black (/mob/living/carbon/human): equipOutfit(Clown (/datum/outfit/job/clown), 0)
   Slippy Joe (/obj/effect/mob_spawn/human/corpse/damaged/legioninfested): equip(Elliot Black (/mob/living/carbon/human), 1)
   Slippy Joe (/obj/effect/mob_spawn/human/corpse/damaged/legioninfested): create(null, 1, null)
   Slippy Joe (/obj/effect/mob_spawn/human/corpse/damaged/legioninfested): Initialize(0)

[2019-09-01T23:39:27] Runtime in magic.dm,81: undefined proc or verb /mob/dead/observer/revive(). 
   proc name: on hit (/obj/item/projectile/magic/resurrection/on_hit)
   usr: Agamemnon (mrmagolor) (/mob/living/carbon/human)
   usr.loc: The floor (144,129,1) (/turf/simulated/floor/plasteel)
   src: the bolt of resurrection (/obj/item/projectile/magic/resurrection)
   src.loc: the floor (142,129,1) (/turf/simulated/floor/plasteel)
   call stack:
   the bolt of resurrection (/obj/item/projectile/magic/resurrection): on hit(Ceres (/mob/dead/observer), 0, "chest")

**Comment by MrMagolor** [comment]
Bugs I noticed in the testmerge:

Inserting robobrains (presumably MMIs/posibrains as well) into borgs results in making a dead borg with the name of "Default (job pref name here)-0" while leaving the exoskeleton behind. Roundstart borgs are not affected.

Trying to open up a wires menu results in: nothing.

**Comment by Fox-McCloud** [comment]
So...this PR is cancer.

I don't think I'll be able to selectively fix things.

**Comment by Fox-McCloud** [comment]
What i mean to say is, uh, I originally meant for this to just address a few Initialization to "tide us over".

I don't think I can do that---I'm going to have to convert over like 99% of the initializations.

File: code/game/machinery/camera/presets.dm [hunk 1]
```diff
@@ -35,19 +35,23 @@
...
-				var/area/CA = get_area(C)
-				if(CA.type == A.type)
-					if(C.number)
-						number = max(number, C.number+1)
-			c_tag = "[A.name] #[number]"
+/obj/machinery/camera/autoname/Initialize(mapload)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/camera/autoname/Initialize(mapload)
```
Comment (Fox-McCloud) [inline code/game/machinery/camera/presets.dm]: Uhhhhhhhh no.` LateInitialize`....

File: code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm [hunk 2]
```diff
@@ -81,7 +81,7 @@
 	loot = list()
 	butcher_results = list(/obj/item/stack/ore/diamond = 2, /obj/item/stack/sheet/sinew = 2, /obj/item/stack/sheet/bone = 1)
 
-/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/Initialize()
+/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/random/Initialize(mapload)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/mining/basilisk.dm]: Change back.

File: code/modules/mob/living/simple_animal/hostile/mining/hivelord.dm [hunk 3]
```diff
@@ -169,7 +169,7 @@
 	var/dwarf_mob = FALSE
 	var/mob/living/carbon/human/stored_mob
 
-/mob/living/simple_animal/hostile/asteroid/hivelord/legion/Initialize()
+/mob/living/simple_animal/hostile/asteroid/hivelord/legion/random/Initialize(mapload)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/mining/hivelord.dm]: Change back.

**Comment by AffectedArc07** [comment]
An F for `SSatoms` and the work it will soon undertake

**Comment by Fox-McCloud** [comment]
Yeah, I'm going to close this. This still need to be done, but there's bigger things that need to be refactored/handled. 

Won't delete the branch though in case I want to revive it or someone else wants to take it over.

**Pull request closed without merging**